src/ProtocolGateway.Core/Instrumentation/IPerformanceCounterManager.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.ProtocolGateway.Instrumentation
+{
+    public interface IPerformanceCounterManager
+    {
+        SafePerformanceCounter GetCounter(string category, string name);
```

Comment:
>SafePerformanceCounter [](start = 8, length = 22)

can we make this IPerformanceCounter and make SafePerformanceCounter an implementation detail of Win-only version?